### PR TITLE
Warn on multi-line 'message' in semgrep rules

### PR DESCRIPTION
**Pull request by huonw**
This adds a rule `yaml/semgrep/multi-line-message` that flags multi-line messages in semgrep rules, identified as suboptimal in #1431. As discussed in #1431, this may actually be better addressed in semgrep itself, rather than a rule/convention. This is done in the first commit e08ebd8.

It also rewrites all existing rules that use `message: |` to use `message: >-` instead (mechanically: `sed -i '' 's/message: |/message: >-/g' **/*.yaml`), in the second commit 776d488.

### CODE REVIEW

File: yaml/semgrep/multi-line-message.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+- id: multi-line-message
+  message: >-
+    This rule has a multi-line message field, which may display poorly
+    in a terminal. Consider ensuring it is on one line. For example,
+    use `message: >-`, not `message: |`.
+  languages: [yaml]
+  patterns:
+  - pattern-inside: 'rules: [..., $RULE, ...]'
+  - pattern: |
+      message: "=~/[\\n\\r]/"
```
Comment (huonw) [inline yaml/semgrep/multi-line-message.yaml:11]: I couldn't work out the right escaping to express this with metavariable-regex, so I've gone with the deprecated string matching syntax instead.

**Comment by minusworld** [comment]
Cool, thanks! I'm gonna skim over all the changes just to make sure there aren't any adverse effects, then we can get this rule in 👍

**Review by minusworld — CHANGES_REQUESTED** [review]
Looks like there's a few rules that broke: https://github.com/returntocorp/semgrep-rules/pull/1432/checks?check_run_id=3447556374

**Comment by huonw** [comment]
Ah, thanks for starting CI. I'll look at fixing those sometime over the next day or two.

**Comment by huonw** [comment]
I think I've addressed most of them now.

**Comment by minusworld** [comment]
Huh, weird that `test-develop` is failing. I'll have to investigate further. It's not required but I wanna double-check to make sure it doesn't break things in the next release. Aside from that, this PR is pretty much done. Thanks! 😄

**Comment by huonw** [comment]
Great! 

I'm sure you'll find this quickly enough, but it looks like other builds are failing with a similar `cp: can't stat '/src/semgrep_runs_output.tar.gz': No such file or directory` error in `test-develop` too, e.g. #1440.

**Comment by minusworld** [comment]
@huonw OK, we've made some progress here - the ❌ 's are now a result of this branch being out-of-sync with the `develop` branch upstream. If you merge the develop branch in here, I think that will fix the problem. Thanks for your patience 😅

**Comment by huonw** [comment]
I've merged and updated the messages of a few extra rules that have landed in develop recently.

**Review by minusworld — APPROVED** [review]

**Comment by minusworld** [comment]
Thanks @huonw !

**Pull request merged**